File: Node/Euler008.js
#!/usr/bin/env node
var fs = require('fs');

function main() {
  const factorCt = 13;

  var contents = fs.readFileSync('input/Euler008.txt', 'utf8');
  contents = contents
    .split('')
    .filter((c) => c >= '0' && c <= '9')
    .map((c) => parseInt(c));

  // Keep track of the last n digits in this circular buffer.
  buffer = Array(factorCt);
  buffer.fill(0);

  var product = 0;
  contents.forEach((c, index) => {
    buffer[index % factorCt] = c;
    product = Math.max(product, buffer.reduce((collector, value) => collector * value));
  })
  console.log(product);
}

if (require.main === module) {
  main();
}

File: Kotlin/Euler008.kt
package Euler008

import java.io.File
import java.lang.Math

fun main() {
  val factorCt = 13
  var product = 1L

  var buffer = LongArray(factorCt, { 0L })

  var fileContents = File("input/Euler008.txt").readText().asSequence().filter(Character::isDigit).map(Character::getNumericValue).map { i -> i.toLong() }

  fileContents.withIndex().forEach { it ->
    buffer[it.index % factorCt] = it.value
    product = Math.max(product, buffer.reduce { a, b -> a * b })
  }

  println(product)
}
